### PR TITLE
action-items-deletion

### PR DESCRIPTION
**Pull request by romatowskak**
DELETING ACTION ITEMS:

- displaying an action item details
- opening a modal with confirmation message regarding an action item deletion
- removing an action item from the list of all action items
- refreshing the view
- displaying again a message telling to select an action item to view details

### CODE REVIEW

File: src/app/action-items/action-items.component.ts [hunk 1]
```diff
@@ -21,9 +21,10 @@ export class ActionItemsComponent implements OnInit, OnDestroy {
   isLoadingActionItems = false;
   isLoadingActionItem = false;
   actionItemId: string | undefined;
-  actionItem: ActionItem;
+  actionItem: ActionItem | undefined;
```
Comment (odudycz) [inline src/app/action-items/action-items.component.ts]:
```suggestion
  actionItem?: ActionItem;
```

File: src/app/action-items/action-items.component.ts [hunk 2]
```diff
@@ -63,6 +64,7 @@ export class ActionItemsComponent implements OnInit, OnDestroy {
           this.retrieveActionItems();
           this.justAddedItemId = item.id;
           this.router.navigate(['/items'], { queryParams: { id: this.justAddedItemId } });
+          this.scrollToLastItem = true;
```
Comment (odudycz) [inline src/app/action-items/action-items.component.ts]: **FIX:** This can't be the last item, as if the application is used by multiple users then at the same time another user may add an action item and after refreshing the list this item will be the last one. You'd need to use id.

File: src/app/confirmationDialog/confirmation-dialog/confirmation-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<div class="container">
+    <div mat-dialog-content class="confirmationMessage">
+        Do you confirm the deletion of this Action Item?
```
Comment (odudycz) [inline src/app/confirmationDialog/confirmation-dialog/confirmation-dialog.component.html]: **SIDENOTE:** Normally all the texts in the applications are not putting text in the templates, but using the translation mechanism, you can check eg. here https://www.codeandweb.com/babeledit/tutorials/how-to-translate-your-angular8-app-with-ngx-translate (you don't need to do that in this PR).

File: src/app/confirmationDialog/confirmation-dialog/confirmation-dialog.component.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+    </div>
+    <div mat-dialog-actions class="buttons">
+        <button class="dialogButton onNoClick" mat-button (click)="onNoClick()">No</button>
```
Comment (odudycz) [inline src/app/confirmationDialog/confirmation-dialog/confirmation-dialog.component.html]: **FIX:** You can use `mat-dialog-close` attribute here.

File: src/app/services/tasksService/tasks.service.ts [hunk 5]
```diff
@@ -120,6 +120,22 @@ export class TasksService {
...
+    setTimeout(() => {
+      this.dataTable.filter(item => {
+        if (item.id === itemId) {
+          this.dataTable.splice(this.dataTable.indexOf(item), 1);
+        }
+      }, 1000);
+    });
+  }
+  editActionItem(itemId: string, itemTitle: string): void {
+    this.dataTable.filter(item => {
```
Comment (odudycz) [inline src/app/services/tasksService/tasks.service.ts]: **FIX:** This also need to use `setTimeout`

File: src/app/edit-item/edit-item/edit-item.component.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import { TasksService } from './../../services/tasksService/tasks.service';
```
Comment (odudycz) [inline src/app/edit-item/edit-item/edit-item.component.ts]: **FIX:** You can reuse the create component. At this stage, they shouldn't differ.

File: src/app/add-item/add-item.component.html [hunk 7]
```diff
@@ -27,7 +28,12 @@ <h2 class="dialogTitle" mat-dialog-title>{{ title }}</h2>
         </mat-form-field>
     </mat-dialog-content>
 </div>
-<mat-dialog-actions class="buttons">
+<mat-dialog-actions class="buttons" *ngIf="createDialog">
     <button class="matRaisedButton close" type="button" mat-button mat-dialog-close>Close</button>
     <button class="matRaisedButton create" [disabled]="dialogForm.invalid" (click)="createActionItem()">Create<span class="faButton" *ngIf="isCreatingActionItem"><i class="fa fa-spinner fa-spin"></i></span></button>
+</mat-dialog-actions>
+
+<mat-dialog-actions class="buttons" *ngIf="!createDialog">
```
Comment (odudycz) [inline src/app/add-item/add-item.component.html]: **SUGGESTION:** This could be merged into single html and use `Save` method that underneath in TypeScript code depending on the inputs will decide if it's Create or Edit mode.

File: src/app/add-item/add-item.component.html [hunk 8]
```diff
@@ -1,5 +1,6 @@
 <div class="container dialogModal">
-    <h2 class="dialogTitle" mat-dialog-title>{{ title }}</h2>
+    <h2 *ngIf="createDialog" class="dialogTitle" mat-dialog-title>{{ dialogTitleForAddingItems }}</h2>
```
Comment (odudycz) [inline src/app/add-item/add-item.component.html]: **SUGGESTION:** We could unify that into single html and single title value and place the if logic in the TypeScript code.

File: src/app/add-item/add-item.component.ts [hunk 9]
```diff
@@ -12,22 +12,36 @@ import { first } from 'rxjs/operators';
 })
 export class AddItemComponent implements OnInit {
   dialogForm: FormGroup;
-  title = 'Create Action Item';
-  projects: Project[];
-  isCreatingActionItem = false;
+  createDialog: boolean;
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: **SUGGESTION:** You could create enum like `ActionItemDialogMode` with values `Create` and `Update` and pass it as the input instead of this boolean. Then based on that you could have the single title value etc.

File: src/app/add-item/add-item.component.ts [hunk 10]
```diff
@@ -12,22 +12,36 @@ import { first } from 'rxjs/operators';
 })
 export class AddItemComponent implements OnInit {
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: **SUGGESTION:** As it's not only adding now but also updating then maybe it'd be worth to name it eg. `ManageActionItemComponent` or `AddOrUpdateActionItemComponent`

File: src/app/add-item/add-item.component.ts [hunk 11]
```diff
@@ -12,22 +12,36 @@ import { first } from 'rxjs/operators';
...
-      .subscribe(projects => (this.projects = projects));
+      .subscribe(projects => {
+        this.projects = projects;
+      });
+
     this.createForm();
+    if (this.data) {
+      this.dialogForm.patchValue({ name: this.data.title, project: this.data.projectName, dueDate: this.data.dueDate });
+      this.id = this.data.id;
+      this.createDialog = this.data.createDialog;
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: **CONCERN:** Could you elaborate on why do we need this assignment here?

File: src/app/add-item/add-item.component.ts [hunk 12]
```diff
@@ -12,22 +12,36 @@ import { first } from 'rxjs/operators';
...
     this.dialogProjects
       .getProjectsNames()
       .pipe(first())
-      .subscribe(projects => (this.projects = projects));
+      .subscribe(projects => {
+        this.projects = projects;
+      });
+
     this.createForm();
+    if (this.data) {
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: **CONCERN:** Why do we need this if?

File: src/app/add-item/add-item.component.ts [hunk 13]
```diff
@@ -58,11 +72,22 @@ export class AddItemComponent implements OnInit {
...
       completed: '0',
       dueDate: dueDate.value
     };
     return newItem;
   }
+  editItem(): void {
+    this.isEditingActionItem = true;
+    const editedItemTitle = this.dialogForm.get('name')!.value;
+    const editedItemProject = this.dialogForm.get('project')!.value;
+    const editedItemDueDate = this.dialogForm.get('dueDate')!.value;
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: **FIX:** You can reuse item preparation code from the create action item - this should be the same.

File: src/app/add-item/add-item.component.ts [hunk 14]
```diff
@@ -58,11 +72,22 @@ export class AddItemComponent implements OnInit {
...
-      projectName: projectName.value.name,
+      projectName: projectName.value,
       type: 'General',
       completed: '0',
       dueDate: dueDate.value
     };
     return newItem;
   }
+  editItem(): void {
+    this.isEditingActionItem = true;
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: **CONCERN:** Why is it needed to assign here this value? Shouldn't it be known already?

File: src/app/item-details/item-details.component.ts [hunk 15]
```diff
@@ -10,16 +13,55 @@ import { Component, Input, ChangeDetectionStrategy, OnChanges } from '@angular/c
...
+    dialogRef.afterClosed().subscribe(res => {
+      if (res) {
+        this.tasksService.deleteActionItem(this.id);
+        this.refreshViewAfterDeletion.emit();
+      }
+    });
+  }
+  openEditDialog(): void {
+    const dialogConfig = new MatDialogConfig();
+    dialogConfig.data = {
```
Comment (odudycz) [inline src/app/item-details/item-details.component.ts]: **SUGGESTION:** It might be worth to consider to move the default dialog config somewhere (eg. to const) and reuse it here (I mean especially width, height, disableClose etc.)

File: src/app/item-details/item-details.component.ts [hunk 16]
```diff
@@ -10,16 +13,55 @@ import { Component, Input, ChangeDetectionStrategy, OnChanges } from '@angular/c
...
+      width: '470px',
+      height: 'auto',
+      disableClose: true,
+      data: {
+        title: !!this.item ? this.item.title : '',
```
Comment (odudycz) [inline src/app/item-details/item-details.component.ts]: **CONCERN:** Why aren't we passing here whole `this.item`?

File: src/app/services/tasksService/tasks.service.ts [hunk 17]
```diff
@@ -30,63 +30,63 @@ export class TasksService {
...
+      id: '7'
     },
     {
       title: 'Cleaning and Organising Your Computer',
       projectName: 'CASD Wilson & Lamberton Middle Schools',
       type: 'Clash',
       completed: '0',
       dueDate: new Date('2020/01/05'),
-      id: this.itemId()
+      id: '8'
```
Comment (odudycz) [inline src/app/services/tasksService/tasks.service.ts]: **FIX:** This shouldn't have changed, we should still generate the unique id instead of hardcoded value.

File: src/app/services/tasksService/tasks.service.ts [hunk 18]
```diff
@@ -117,6 +117,38 @@ export class TasksService {
         } else {
           observer.error(new HttpResponse({ status: 404, statusText: 'No item with such ID found!' }));
         }
+      }, 2000);
+    });
+  }
+  deleteActionItem(itemId: string): void {
```
Comment (odudycz) [inline src/app/services/tasksService/tasks.service.ts]: **FIX:** Please return observable accordingly as in the other methods..

File: src/app/services/tasksService/tasks.service.ts [hunk 19]
```diff
@@ -117,6 +117,38 @@ export class TasksService {
...
+    setTimeout(() => {
+      this.dataTable.filter(item => {
+        if (item.id === itemId) {
+          this.dataTable.splice(this.dataTable.indexOf(item), 1);
+        }
+      }, 1000);
+    });
+  }
+  editActionItem(
```
Comment (odudycz) [inline src/app/services/tasksService/tasks.service.ts]: **CONCERN:** In my opinion, we should pass here the `EditActionItemModel` similarly as in the add action item. Then you could benefit and simplify the code with spread operator. It's also easier to extend the class instead of adding additional parameter (and imho more readable).

File: src/app/add-item/add-item.component.ts [hunk 20]
```diff
@@ -10,59 +10,93 @@ import { first } from 'rxjs/operators';
...
   dialogForm: FormGroup;
-  title = 'Create Action Item';
-  projects: Project[];
-  isCreatingActionItem = false;
+  createDialog: boolean;
+  dialogTitle: string;
+  dialogActionButton: string;
+  projects?: Project[];
+  isSavingDialogData: boolean = false;
+  id: string;
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: **QUESTION:** Why do we need this variable?

File: src/app/add-item/add-item.component.ts [hunk 21]
```diff
@@ -10,59 +10,93 @@ import { first } from 'rxjs/operators';
...
+    public dialogRef: MatDialogRef<AddOrUpdateActionItemComponent>,
     private formBuilder: FormBuilder,
     private dialogProjects: ProjectsService,
-    private tasksService: TasksService
+    private tasksService: TasksService,
+    @Inject(MAT_DIALOG_DATA) public data: any
   ) {}
+
   ngOnInit() {
+    if (this.data.createDialog === true) {
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: If you used the enum eg. `ActionItemDialogMode` with values `Create` and `Edit` then you could shorten this to:
```
this.dialogTitle = `${dialogMode} Action Item`;
this.dialogActionButton = dialogMode.toString();
```

File: src/app/add-item/add-item.component.ts [hunk 22]
```diff
@@ -10,59 +10,93 @@ import { first } from 'rxjs/operators';
...
       .getProjectsNames()
       .pipe(first())
-      .subscribe(projects => (this.projects = projects));
+      .subscribe(projects => {
+        this.projects = projects;
+        this.loaderVisible = false;
+      });
     this.createForm();
+    this.dialogForm.patchValue({
+      title: this.data.item ? this.data.item.title : '',
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: can't we just do `this.dialogForm.updateValue(this.data.item)` ord `this.dialogForm.updateValue(this.data.item || {})`?

File: src/app/add-item/add-item.component.ts [hunk 23]
```diff
@@ -10,59 +10,93 @@ import { first } from 'rxjs/operators';
...
+      projectName: ['', Validators.required],
       dueDate: '',
-      description: ''
+      description: '',
+      id: ''
     });
   }
+  saveForm(): void {
+    this.isSavingDialogData = true;
+    this.id ? this.editItem() : this.createActionItem();
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: **SUGGESTION:** This if should be based on the enum mentioned in https://github.com/romatowskak/examp-ng-app/pull/4/files#r370506748

File: src/app/add-item/add-item.component.ts [hunk 24]
```diff
@@ -10,59 +10,93 @@ import { first } from 'rxjs/operators';
...
       title: title.value,
-      projectName: projectName.value.name,
+      projectName: projectName.value,
       type: 'General',
       completed: '0',
       dueDate: dueDate.value
     };
     return newItem;
   }
+  editItem(): void {
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: **SUGGGESTION**: It's generally good to have a convention of the naming - `createActionItem` vs `editItem`. We should name it either all with or without Action word.

File: src/app/action-items/action-items.component.ts [hunk 25]
```diff
@@ -20,10 +26,14 @@ export class ActionItemsComponent implements OnInit, OnDestroy {
...
-  actionItemId: string | undefined;
-  actionItem: ActionItem;
-  errorMessage: string | undefined;
-  justAddedItemId: string;
+  actionItemId?: string;
+  actionItem?: ActionItem;
+  errorMessage?: string;
+  actionItemIdForScroll: string;
+  scrollToActionItem: boolean;
+  createDialog: boolean;
```
Comment (odudycz) [inline src/app/action-items/action-items.component.ts]: **FIX:** This variable is not used besides the opening dialog. You could safely remove it and pass there proper mode.

File: src/app/action-items/action-items.component.ts [hunk 26]
```diff
@@ -20,10 +26,14 @@ export class ActionItemsComponent implements OnInit, OnDestroy {
...
   isLoadingActionItems = false;
   isLoadingActionItem = false;
-  actionItemId: string | undefined;
-  actionItem: ActionItem;
-  errorMessage: string | undefined;
-  justAddedItemId: string;
+  actionItemId?: string;
+  actionItem?: ActionItem;
+  errorMessage?: string;
+  actionItemIdForScroll: string;
```
Comment (odudycz) [inline src/app/action-items/action-items.component.ts]: **SUGGESTION:** I think that if you made this variable optional then you could get rid off additional [scrollToActionItem variable](https://github.com/romatowskak/examp-ng-app/pull/4/files#diff-697b1334bd6e35d971c84803ccbe914aR33) by clearing this variable in the [line below](https://github.com/romatowskak/examp-ng-app/pull/4/files#diff-697b1334bd6e35d971c84803ccbe914aR92).
```suggestion
  actionItemIdForScroll?: string;
```

File: src/app/add-item/add-item.component.ts (hunk 9 above)
Comment (odudycz) [reply src/app/add-item/add-item.component.ts]: I think that it would be better if you remove this variable and just pass the enum.

File: src/app/add-item/add-item.component.ts [hunk 27]
```diff
@@ -1,68 +1,93 @@
...
+    public dialogRef: MatDialogRef<AddOrUpdateActionItemComponent>,
     private formBuilder: FormBuilder,
     private dialogProjects: ProjectsService,
-    private tasksService: TasksService
+    private tasksService: TasksService,
+    @Inject(MAT_DIALOG_DATA) public data: any
   ) {}
+
   ngOnInit() {
+    this.data.createDialog === true ? (this.dialogMode = dialogMode.create) : (this.dialogMode = dialogMode.edit);
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: See comment [above](https://github.com/romatowskak/examp-ng-app/pull/4/files#r371098973).

File: e2e/src/addAndDisplayDetails/addAndDisplayDetails.e2e-spec.ts [hunk 28]
```diff
@@ -7,9 +7,9 @@ describe('workspace-project App', () => {
     page = new AppPage();
   });
 
-  it('should display table with 8 action items', () => {
+  it('should display table with multiple', () => {
```
Comment (mdybich) [inline e2e/src/addAndDisplayDetails/addAndDisplayDetails.e2e-spec.ts]: `multiple` what? ;)

File: src/app/action-items/action-items.component.html [hunk 29]
```diff
@@ -25,10 +25,14 @@
             <i class="fa fa-spinner fa-spin"></i>
         </div>
         <div class="tBody" *ngIf="!isLoadingActionItems">
+            <div *ngIf="showImageWhenNoActionItem" class="emptyListOfItem">
+                <img class="noItem" src="/assets/empty-inbox.svg">
```
Comment (mdybich) [inline src/app/action-items/action-items.component.html]: I would suggest to use `mat-icon` component and `IconRegistry` to display svg https://material.angular.io/components/icon/overview#svg-icons

File: src/app/action-items/action-items.component.ts [hunk 30]
```diff
@@ -20,10 +26,12 @@ export class ActionItemsComponent implements OnInit, OnDestroy {
   dataSource: ActionItemMapped[];
   isLoadingActionItems = false;
   isLoadingActionItem = false;
-  actionItemId: string | undefined;
-  actionItem: ActionItem;
-  errorMessage: string | undefined;
-  justAddedItemId: string;
+  actionItemId?: string;
```
Comment (odudycz) [inline src/app/action-items/action-items.component.ts]: **CONCERN:** Do we need both this variable? Can't we use `actionItem?.id` instead?

File: src/app/action-items/action-items.component.ts [hunk 31]
```diff
@@ -38,8 +46,8 @@ export class ActionItemsComponent implements OnInit, OnDestroy {
   ngOnInit() {
     this.retrieveActionItems();
     this.subscribeToQueryParams();
+    this.dialogData = { width: '470px', height: 'auto', disableClose: true };
```
Comment (odudycz) [inline src/app/action-items/action-items.component.ts]: **SUGGESTION:** This could set as const outside of component

File: src/app/action-items/action-items.component.ts [hunk 32]
```diff
@@ -20,10 +26,12 @@ export class ActionItemsComponent implements OnInit, OnDestroy {
   dataSource: ActionItemMapped[];
   isLoadingActionItems = false;
   isLoadingActionItem = false;
-  actionItemId: string | undefined;
-  actionItem: ActionItem;
-  errorMessage: string | undefined;
-  justAddedItemId: string;
+  actionItemId?: string;
+  actionItem?: ActionItem;
```
Comment (odudycz) [inline src/app/action-items/action-items.component.ts]: **SUGGESTION:** General note - please make sure that all fields have the proper modifiers (so if something is not used in template then it's private).

File: src/app/action-items/action-items.component.ts [hunk 33]
```diff
@@ -49,26 +57,30 @@ export class ActionItemsComponent implements OnInit, OnDestroy {
       .subscribe(tasks => {
         this.dataSource = tasks;
         this.isLoadingActionItems = false;
-        this.isLoadingActionItem = false;
+        this.showImageWhenNoActionItem = false;
+        if (tasks.length === 0) this.showImageWhenNoActionItem = true;
```
Comment (odudycz) [inline src/app/action-items/action-items.component.ts]:
```suggestion
         this.showImageWhenNoActionItem = tasks.length === 0;
```

File: src/app/action-items/action-items.component.ts [hunk 34]
```diff
@@ -49,26 +57,30 @@ export class ActionItemsComponent implements OnInit, OnDestroy {
...
           this.retrieveActionItems();
-          this.justAddedItemId = item.id;
-          this.router.navigate(['/items'], { queryParams: { id: this.justAddedItemId } });
+          this.actionItemIdForScroll = item.id;
+          this.router.navigate(['/items'], { queryParams: { id: this.actionItemIdForScroll } });
         }
       });
   }
-  private triggerScrollTo() {
+  private triggerScrollTo(itemId: string): void {
```
Comment (odudycz) [inline src/app/action-items/action-items.component.ts]: **CONCERN:** Shouldn't we clear value of `this.actionItemIdForScroll` after scrolling?

File: src/app/add-item/add-item.component.ts [hunk 35]
```diff
@@ -1,68 +1,94 @@
...
+    public dialogRef: MatDialogRef<AddOrUpdateActionItemComponent>,
     private formBuilder: FormBuilder,
     private dialogProjects: ProjectsService,
-    private tasksService: TasksService
+    private tasksService: TasksService,
+    @Inject(MAT_DIALOG_DATA) public data: any
   ) {}
+
   ngOnInit() {
+    !this.data.item ? (this.dialogMode = dialogMode.create) : (this.dialogMode = dialogMode.edit);
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: **CONCERN:** Why do we need this conditional statement and can't just assign the value?

File: src/app/add-item/add-item.component.ts (hunk 24 above)
Comment (odudycz) [reply src/app/add-item/add-item.component.ts]: still not applied

File: src/app/action-items/action-items.component.html [hunk 36]
```diff
@@ -39,16 +43,17 @@
...
                 <div class="flexRow bodyCell inner" role="cell">
                     <a><i class="fa fa-edit"></i></a>
                 </div>
             </div>
         </div>
     </div>
-    <div class="innerContainer detailsContainer ">
-        <app-item-details [id]="actionItemId" [item]="actionItem" [isLoadingActionItem]="isLoadingActionItem" [errorMessage]="errorMessage">
+    <div class="innerContainer detailsContainer">
+        <app-item-details [showImageWhenNoActionItem]="showImageWhenNoActionItem" [actionItemId]="actionItemId" [actionItem]="actionItem" [isLoadingActionItem]="isLoadingActionItem" [errorMessage]="errorMessage" [dialogData]="dialogData" (refreshViewAfterDeletion)="refreshViewAfterDeletion($event)"
```
Comment (mdybich) [inline src/app/action-items/action-items.component.html]: `actionItem` already contains `id` - why do you need to pass `actionItem` as well as `actionItemId`?

File: src/app/services/tasksService/tasks.service.ts [hunk 37]
```diff
@@ -1,126 +1,173 @@
...
   title: string;
   projectName: string;
   type?: string;
   completed?: string;
   dueDate?: Date;
+  description?: string;
   id: string;
 }
 
+const localStorageKey = 'localDataTable';
```
Comment (mdybich) [inline src/app/services/tasksService/tasks.service.ts]: This can be moved as a private readonly memeber of the TasksService

File: src/app/services/tasksService/tasks.service.ts [hunk 38]
```diff
@@ -1,126 +1,173 @@
...
         } else {
           observer.error(new HttpResponse({ status: 404, statusText: 'No item with such ID found!' }));
         }
+      }, 2000);
+    });
+  }
+  deleteActionItem(actiontemId: string): Observable<ActionItem[]> {
+    return new Observable(observer => {
+      setTimeout(() => {
+        const deletedItemIndex = this.dataTable.findIndex(x => x.id === actiontemId);
```
Comment (mdybich) [inline src/app/services/tasksService/tasks.service.ts]: You can change this whole logic into one line of code:
```
this.dataTable = this.dataTable.filter(data => data.id !== actionItemId);
this.storage.set(localStorageKey, this.dataTable);
```

File: src/app/add-item/add-item.component.ts [hunk 39]
```diff
@@ -10,59 +10,80 @@ import { first } from 'rxjs/operators';
...
   styleUrls: ['./add-item.component.css']
 })
-export class AddItemComponent implements OnInit {
-  dialogForm: FormGroup;
-  title = 'Create Action Item';
-  projects: Project[];
-  isCreatingActionItem = false;
+export class AddOrUpdateActionItemComponent implements OnInit {
+  private dialogForm: FormGroup;
+  dialogMode: string;
```
Comment (odudycz) [inline src/app/add-item/add-item.component.ts]: **CONCERN:** Why is it string and not enum?
